cmd/labs/project/entrypoint.go

```diff
@@ -0,0 +1,250 @@
+package project
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/internal/build"
+	"github.com/databricks/cli/libs/cmdio"
+	"github.com/databricks/cli/libs/env"
+	"github.com/databricks/cli/libs/log"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/spf13/cobra"
+)
+
+type Entrypoint struct {
+	*Project
+
+	RequireRunningCluster bool `yaml:"require_running_cluster,omitempty"`
+	IsUnauthenticated     bool `yaml:"is_unauthenticated,omitempty"`
+	IsAccountLevel        bool `yaml:"is_account_level,omitempty"`
+	IsBundleAware         bool `yaml:"is_bundle_aware,omitempty"`
+}
+
+var ErrNoLoginConfig = errors.New("no login configuration found")
+var ErrMissingClusterID = errors.New("missing a cluster compatible with Databricks Connect")
+var ErrMissingWarehouseID = errors.New("missing a SQL warehouse")
+var ErrNotInTTY = errors.New("not in an interactive terminal")
+
+func (e *Entrypoint) NeedsCluster() bool {
+	if e.Installer == nil {
+		return false
+	}
+	if e.Installer.RequireDatabricksConnect && e.Installer.MinRuntimeVersion == "" {
+		e.Installer.MinRuntimeVersion = "13.1"
+	}
+	return e.Installer.MinRuntimeVersion != ""
+}
+
+func (e *Entrypoint) NeedsWarehouse() bool {
+	if e.Installer == nil {
+		return false
+	}
+	return len(e.Installer.WarehouseTypes) != 0
+}
+
+func (e *Entrypoint) Prepare(cmd *cobra.Command) (map[string]string, error) {
+	ctx := cmd.Context()
+	libDir := e.EffectiveLibDir(ctx)
+	environment := map[string]string{
+		"DATABRICKS_CLI_VERSION":     build.GetInfo().Version,
+		"DATABRICKS_LABS_CACHE_DIR":  e.CacheDir(ctx),
+		"DATABRICKS_LABS_CONFIG_DIR": e.ConfigDir(ctx),
+		"DATABRICKS_LABS_STATE_DIR":  e.StateDir(ctx),
+		"DATABRICKS_LABS_LIB_DIR":    libDir,
+	}
+	if e.IsPythonProject(ctx) {
+		e.preparePython(ctx, environment)
+	}
+	cfg, err := e.validLogin(cmd)
+	if err != nil {
+		return nil, fmt.Errorf("login: %w", err)
+	}
+	// cleanup auth profile and config file location,
+	// so that we don't confuse SDKs
+	cfg.Profile = ""
+	cfg.ConfigFile = ""
```

Comment:
These won't be cleared in the environment because they are zero. That means they are not returned from `e.environmentFromConfig` below and are not cleared for the subprocess, which could confuse downstream programs.

Reply:
@pietern they won't get to environment :) see `e.environmentFromConfig(cfg)`

Reply:
@pietern the only case possible when we have `--profile` (DATABRICKS_CONFIG_PROFILE) is the local dev machine. Github actions (and likes) won't have the `DATABRICKS_CONFIG_PROFILE` / `DATABRICKS_CONFIG_FILE` envs set, but rather direct ones, like client ids or hosts. there are tests that verify this to work from github actions-like environments